api/engineapi/middleware.py

```diff
@@ -102,3 +102,38 @@ def __init__(
         if internal_error is not None:
             print(internal_error)
             # reporter.error_report(internal_error)
+
+
+class ExtractBearerTokenMiddleware(BaseHTTPMiddleware):
+    """
+    Checks the authorization header on the request and extract token.
+    """
+
+    def __init__(self, app, whitelist: Optional[Dict[str, str]] = None):
+        self.whitelist: Dict[str, str] = {}
+        if whitelist is not None:
+            self.whitelist = whitelist
+        super().__init__(app)
+
+    async def dispatch(
+        self, request: Request, call_next: Callable[[Request], Awaitable[Response]]
+    ):
+        # Filter out endpoints with proper method to work without Bearer token (as create_user, login, etc)
+        path = request.url.path.rstrip("/")
+        method = request.method
+        if path in self.whitelist.keys() and self.whitelist[path] == method:
+            return await call_next(request)
+
+        authorization_header = request.headers.get("authorization")
+        if authorization_header is None:
+            return Response(
+                status_code=403, content="No authorization header passed with request"
+            )
+        user_token_list = authorization_header.split()
```

Comment:
This is not user_token_list, this is components of auth header (bearer and token), better to call it 
`authorization_header_components`